Zend/zend_variables.c

```diff
@@ -71,6 +71,7 @@ static void ZEND_FASTCALL zend_string_destroy(zend_string *str)
 static void ZEND_FASTCALL zend_reference_destroy(zend_reference *ref)
 {
 	ZEND_ASSERT(!ZEND_REF_HAS_TYPE_SOURCES(ref));
+	GC_REMOVE_FROM_BUFFER(ref);
 	i_zval_ptr_dtor(&ref->val);
 	efree_size(ref, sizeof(zend_reference));
```

Comment:
There are some other places where the zend_reference may be deallocated (e.g. zend_assign_to_variable)

Reply:
I'm wondering whether we actually need to handle other places: We are only adding resources to the root buffer during garbage collection here, not before. So references with a root buffer entry shouldn't be visible from normal code.